### PR TITLE
boot: complain about reused asset name during initial install

### PR DESCRIPTION
**Pull request by bboozzoo**
Complain loudly when we see more than one version (content-wise) of a trusted
boot asset installed during initial setup.

### CODE REVIEW

File: boot/assets.go [hunk 1]
```diff
@@ -197,16 +197,19 @@ func isAlreadyTracked(current []*trackedAsset, newAsset *trackedAsset) bool {
 	return false
```
Comment (pedronis) [inline boot/assets.go:197]: this function can be deleted now, no?

File: boot/assets.go [hunk 2]
```diff
@@ -197,16 +197,19 @@ func isAlreadyTracked(current []*trackedAsset, newAsset *trackedAsset) bool {
...
 }
 
-func trackedAssetsToBootMap(trackedAssets []*trackedAsset) bootAssetsMap {
-	if len(trackedAssets) == 0 {
-		return nil
+func isAlreadyTrackedInBootMap(bam bootAssetsMap, newAsset *trackedAsset) bool {
```
Comment (pedronis) [inline boot/assets.go:200]: isAssetAlreadyTracked

File: boot/assets.go [hunk 3]
```diff
@@ -197,16 +197,19 @@ func isAlreadyTracked(current []*trackedAsset, newAsset *trackedAsset) bool {
...
+	return isAssetHashTrackedInBootMap(bam, newAsset.name, newAsset.hash)
+}
+
+func isAssetHashTrackedInBootMap(bam bootAssetsMap, assetName, assetHash string) bool {
```
Comment (pedronis) [inline boot/assets.go:204]: isAssetHashTrackedInMap

**Review by pedronis — APPROVED** [review]
thanks, some small comments about unused functions and names

**Comment by bboozzoo** [comment]
@mvo5 thanks for pushing the fixes!

**Pull request merged**